### PR TITLE
cache implementation & Exception handling 

### PR DESCRIPTION
**Pull request by kssree**
updated code

### CODE REVIEW

File: app-shopping/src/main/java/com/shopping/controller/PhoneController.java [hunk 1]
```diff
@@ -72,5 +74,17 @@ public Optional<PhoneModel> getById(@PathVariable Long id) throws ProductNotFoun
         }
         return res;
     }
+    @PutMapping("/savemodelno")
+    public ResponseEntity<String> update(@RequestBody  PhoneModel pm)
```
Comment (puvvadakrishna) [inline app-shopping/src/main/java/com/shopping/controller/PhoneController.java:78]: what is the difference b/w put and patch?

why this is used here?

File: app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java [hunk 2]
```diff
@@ -47,4 +49,23 @@ private PhoneModel updateColor(PhoneModel modle) {
         return modle;
     }
 
+    public Long addPhone(PhoneModel pm)
+    {
+        return modelRepository.save(pm).getId();
+    }
+
+    public void deletePhone(String modelPrice)
```
Comment (puvvadakrishna) [inline app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java:57]: use @Transactional on top of the deletePhone method

File: app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java [hunk 3]
```diff
@@ -47,4 +49,23 @@ private PhoneModel updateColor(PhoneModel modle) {
...
+    {
+        modelRepository.deleteBymodelPrice(modelPrice);
+    }
+
+    @Override
+    public List<PhoneModel> findByModelColour(String modelColour) {
+        List<PhoneModel> pm = modelRepository.findByModelColour(modelColour);
+        if (!pm.isEmpty()) {
```
Comment (puvvadakrishna) [inline app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java:65]: use pm.isPresent()

File: app-shopping/src/main/java/com/shopping/controller/PhoneController.java (hunk 1 above)
Comment (kssree) [reply app-shopping/src/main/java/com/shopping/controller/PhoneController.java:78]: Patch is used to update the resource while Put is used to update particular entry of resource for example email address...
Its a mistake ill look into it and do the required changes. Thank you

File: app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java (hunk 2 above)
Comment (kssree) [reply app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java:57]: Thanks for the input, delete is working now.

File: app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java (hunk 3 above)
Comment (kssree) [reply app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java:65]: If i use Optional<List<PhoneModel>> its returning the **Optional[[]]**  value and showing some value is present and going to the if block even if the modelColour is not present.

File: app-shopping/src/main/java/com/shopping/controller/PhoneController.java (hunk 1 above)
Comment (puvvadakrishna) [reply app-shopping/src/main/java/com/shopping/controller/PhoneController.java:78]: I think its visevera.
Put is used to update the resource.
Patch is to update particular entry of resource for example email address...

https://www.google.com/search?q=difference+between+put+and+patch&rlz=1C5GCCM_en&oq=difference+between+pu&gs_lcrp=EgZjaHJvbWUqBwgBEAAYgAQyBggAEEUYOTIHCAEQABiABDIHCAIQABiABDIHCAMQABiABDIHCAQQABiABDIHCAUQABiABDIHCAYQABiABDIHCAcQABiABDIHCAgQABiABDIHCAkQABiABNIBCDY3MzZqMGo3qAIAsAIA&sourceid=chrome&ie=UTF-8

**Pull request merged**